doc/trapezoidal_industrial_planner/trapezoidal_industrial_planner.rst

```diff
@@ -0,0 +1,366 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- ``planner_id``: PTP
+- ``group_name``: name of the planning group
+- ``max_velocity_scaling_factor``: scaling factor of maximal joint velocity
+- ``max_acceleration_scaling_factor``: scaling factor of maximal joint acceleration/deceleration
+- ``start_state/joint_state/(name, position and velocity``: joint name/position/velocity(optional) of the start state.
+- ``goal_constraints`` (goal can be given in joint space or Cartesian space)
+- for a goal in joint space
+    - ``goal_constraints/joint_constraints/joint_name``: goal joint name
+    - ``goal_constraints/joint_constraints/position``: goal joint position
+- for a goal in Cartesian space
+    - ``goal_constraints/position_constraints/header/frame_id``: frame this data is associated with
+    - ``goal_constraints/position_constraints/link_name``: target link name
+    - ``goal_constraints/position_constraints/constraint_region``: bounding volume of the target point
+    - ``goal_constraints/position_constraints/target_point_offset``: offset (in the link frame) for the target point on
+      the target link (optional)
+- ``max_velocity_scaling_factor``: rescale the maximal velocity
+- ``max_acceleration_scaling_factor``: rescale the maximal acceleration/deceleration
+
+
+Planning results in ``moveit_msg::MotionPlanResponse``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  ``trajectory_start``: first robot state of the planned trajectory
+-  ``trajectory/joint_trajectory/joint_names``: a list of the joint
+   names of the generated joint trajectory
+-  ``trajectory/joint_trajectory/points/(positions,velocities,accelerations,time_from_start)``:
+   a list of generated way points. Each point has
+   positions/velocities/accelerations of all joints (same order as the
+   joint names) and time from start. The last point will have zero
+   velocity and acceleration.
+-  ``group_name``: name of the planning group
+-  ``error_code/val``: error code of the motion planning
+
+The LIN motion command
+----------------------
+
+This planner generates linear Cartesian trajectory between goal and
+start poses. The planner uses the Cartesian limits to generate a
+trapezoidal velocity profile in Cartesian space. The translational
+motion is a linear interpolation between start and goal position vector.
+The rotational motion is quaternion slerp between start and goal
+orientation. The translational and rotational motion is synchronized in
+time. This planner only accepts start state with zero velocity. Planning
+result is a joint trajectory. The user needs to adapt the Cartesian
+velocity/acceleration scaling factor if the motion plan fails due to
+violation of joint space limits.
+
+Input parameters in ``moveit_msgs::MotionPlanRequest``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  ``planner_id``: LIN
+-  ``group_name``: name of the planning group
+-  ``max_velocity_scaling_factor``: scaling factor of maximal Cartesian
```

Comment:
Good question. The implementation scales the cartesian velocity, so that e.g. 1m/s maximum speed is executed with 0.25m/s for a `scaling_factor = 0.25`. Isn't that what you'd expect as a user of the planner?

Reply:
Yes, but the name is the same as the scaling factor the `move_group` applies to each maximum joint velocity.

In cartesian space, I would agree if it was called `max_cartesian_velocity_scaling_factor`. But at that point, why not just make it `max_cartesian_velocity`? That's very desirable anyway, I remember it has come up in issues before.